### PR TITLE
chore: move CLI specific login providers

### PR DESCRIPTION
**Pull request by kian99**
This PR is a result of a discussion in #17433 and this [comment](https://github.com/juju/juju/pull/16955#issuecomment-2122317852) from #16955, c/p the main point here,
>This (the session token login provider) shouldn't be in the API package. The API package is just for serializing and deserializing data on and off the wire. Can we move this to the cmd package.

Since #16955, Juju supports the concept of a login provider, where the client can define how information should be presented to the login facade, somewhat decoupling what is put on the wire with how it is captured and passed to the `api` layer.

This PR moves the interactive login providers to a new package in `cmd/juju/loginproviders`. The original suggestion was to move the providers to `cmd/api` or `cmd/internal/api` both of which don't seem to exist but I'm happy to move these to wherever is more appropriate. **Please let me know where is best.**

Another benefit of this is that the `LoginResultParams` now has a constructor method which encapsulates the logic going from a `LoginResult` response from Juju into a `LoginResultParams` which was duplicated in several places.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [ ] Comments saying why design decisions were made
- [ ] ~~Go unit tests, with comments saying what you're testing~~
- [ ] ~~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~~
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

Mechanical/structural changes.

### CODE REVIEW

**Comment by kian99** [comment]
cc @SimonRichardson

File: api/client_credentials_login_provider.go [hunk 1]
```diff
@@ -7,8 +7,6 @@ import (
 	"context"
```
Comment (SimonRichardson) [inline api/client_credentials_login_provider.go:7]: Rename this file.

File: api/userpass_login_provider.go [hunk 2]
```diff
@@ -15,7 +15,6 @@ import (
 	"github.com/juju/featureflag"
```
Comment (SimonRichardson) [inline api/userpass_login_provider.go:15]: Rename this file.

**Review by SimonRichardson — COMMENTED** [review]
Question: why was `cmd/juju/loginprovider` location picked? If anything I'd move it to `cmd/juju/internal`. If JAAS requires access to these, then we need to fully understand your use cases.

Also, can you rebase your commit. We're moving to https://www.conventionalcommits.org/en/v1.0.0/

**Comment by kian99** [comment]
> why was cmd/juju/loginprovider location picked?

This was an arbitrary location. I'll move the files to `cmd/juju/internal`. 
Regarding conventional commits, what type would this be? `chore`?

**Comment by kian99** [comment]
Just to confirm @SimonRichardson, do you want the package to be `internal`? Or should I do `cmd/juju/internal/loginprovider`?

**Comment by kian99** [comment]
Moved CLI specific login providers to `cmd/internal/loginprovider/` after a discussion since the `modelcmd` imports it.

**Review by SimonRichardson — APPROVED** [review]
Thanks for pushing this over the line.

**Pull request merged**